### ISSUE TITLE
Headers not being merged with defaults

### ISSUE THREAD

**Issue by sploders101** [apache/couchdb-nano#273]
## Expected Behavior

When using custom headers (example below), they should be merged with the request headers before the request is sent.

## Current Behavior

When using custom headers (example below), they are not merged with the request headers in changesReader. This results in incorrect headers being sent in the final request.

## Possible Solution

The offending call is in `9.0.3` at `nano/lib/changesReader.js:248`, where Object.assign is used instead of a deepMerge function. This *overwrites* the headers field with the requestDefaults version rather than merging, resulting in missing headers when a request is finally sent. If I replace the offending code with this, it works.
```typescript
const req = Object.assign({
  method: (opts.method || 'GET'),
  headers: headers,
  uri: cfg.url
}, {
  ...cfg.requestDefaults,
  headers: Object.assign(headers, cfg.requestDefaults && cfg.requestDefaults.headers ? cfg.requestDefaults.headers : {}),
})
```

## Steps to Reproduce (for bugs)

1. Set up a couchdb connection
```typescript
export const couchdb = nano({
	requestDefaults: {
		headers: {
			"X-Auth-CouchDB-UserName": "server",
			"X-Auth-CouchDB-Roles": "_admin",
		},
	},
	url: dbURL,
});
```
2. Use a request that utilizes headers. I am using `changesReader`, which fails on start, so that's probably the easiest way to test.

## Context

I am trying to use proxy authentication in CouchDB since I link to an AD server for authentication for SSO. I have set up a simple authentication proxy in my application, but use the code above to create a nano instance usable by the server for privileged tasks.

## Your Environment

* Version used: 9.0.3
* Browser Name and version: N/A
* Operating System and version (desktop or mobile): Mac OS 11.4
* Link to your project: Private repository

**Comment by sploders101** [comment]
@glynnbird Thank you for being on top of this! I would have opened a PR, but I was trying to hit a deadline at the time. I tried to compromise by including my patches in the issues (I used `patch-package` to get a working version into production, because I wanted to use the `changesReader` to avoid re-implementing it under the EventSource protocol, and it's working great! I'd highly recommend it if you're ever having to work past bugs in another package). Looking forward to the update!
